lib/mix/phoenix/schema.ex

```diff
@@ -425,7 +425,7 @@ defmodule Mix.Phoenix.Schema do
       mix phx.gen.schema Comment comments body:text status:enum:published:unpublished
   """
 
-  defp validate_attr!({name, :datetime}), do: validate_attr!({name, :naive_datetime})
+  defp validate_attr!({name, :datetime}), do: {name, :naive_datetime}
```

Comment:
I think the intent with the datetime case may be to throw if (for whatever reason) the naive_datetime type was removed from the `@valid_types`.

Reply:
@SteffenDE, Thank you for the review! 🙇🏼 I thought the same. Though I am not fun of extra app code processing to catch possible future bugs. Tests are better approach for it. And conveniently there is test that will fail if somebody not fully clean up places with `naive_datetime` usage. This is why I think it is safe to remove extra protection in code.
[Existing test for example](https://github.com/phoenixframework/phoenix/blob/main/test/mix/tasks/phx.gen.schema_test.exs#L252)
```elixir
  test "generates schema with proper datetime types", config do
    in_tmp_project config.test, fn ->
      Gen.Schema.run(~w(Blog.Comment comments title:string drafted_at:datetime published_at:naive_datetime edited_at:utc_datetime locked_at:naive_datetime_usec))

      assert_file "lib/phoenix/blog/comment.ex", fn file ->
        assert file =~ "field :drafted_at, :naive_datetime"
        assert file =~ "field :published_at, :naive_datetime"
```